File: test/test_count.cpp
#include <stdio.h>
#include "TRTCContext.h"
#include "DVVector.h"
#include "sequence.h"
#include "count.h"

int main()
{
	int hin[2000];
	for (int i = 0; i < 2000; i++)
		hin[i] = i % 100;

	DVVector din("int32_t", 2000, hin);
	size_t c;
	TRTC_Count(din, DVInt32(47), c);
	printf("%d\n", (int)c);

	TRTC_Sequence(din);
	Functor op = {{},{ "x" }, "        return (x%100)==47;\n" };
	TRTC_Count_If(din, op, c);
	printf("%d\n", (int)c);

	return 0;
}

File: CSharp/test/test_count/test_count.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ThrustRTCSharp;

namespace test_count
{
    class test_count
    {
        static void Main(string[] args)
        {
            int[] hin=new int[2000];
            for (int i = 0; i < 2000; i++)
                hin[i] = i % 100;

            DVVector din = new DVVector(hin);
            Console.WriteLine(TRTC.Count(din, new DVInt32(47)));

            TRTC.Sequence(din);
            Functor op = new Functor(new string[]{ "x" }, "        return (x%100)==47;\n" );
            Console.WriteLine(TRTC.Count_If(din, op));

        }

        
    }
}
